File: UIServer/Areas/CMS/Controllers/Set_SetUpController.cs
using Kaizen.BusinessLogic.CMS;
using Kaizen.BusinessLogic.Configuration;
using Kaizen.CMS;
using Kaizen.Configuration;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace UIServer.Areas.CMS.Controllers
{
    public class Set_SetUpController : Controller
    {
        // GET: Set_SetUp
        public ActionResult Index(string KaizenPublicKey)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            if (KaizenUser == null) return RedirectToAction("LoginToCompany", "SysUser");
            return PartialView();
        }
        public ActionResult StandardFields(string KaizenPublicKey)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            if (KaizenUser == null) return RedirectToAction("LoginToCompany", "SysUser");
            return PartialView();
        }
        public ActionResult SaveData(CM00000 CM00000, string KaizenPublicKey)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            if (KaizenUser == null) return RedirectToAction("LoginToCompany", "SysUser");
            CM00000Services service = new CM00000Services(KaizenUser);
            return Json(service.AddCM00000(CM00000), JsonRequestBehavior.AllowGet);
        }
        public ActionResult UpdateData(CM00000 CM00000, string KaizenPublicKey)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            if (KaizenUser == null) return RedirectToAction("LoginToCompany", "SysUser");
            CM00000Services service = new CM00000Services(KaizenUser);
            return Json(service.Update(CM00000), JsonRequestBehavior.AllowGet);
        }
        public ActionResult DeleteData(CM00000 CM00000, string KaizenPublicKey)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            if (KaizenUser == null) return RedirectToAction("LoginToCompany", "SysUser");
            CM00000Services service = new CM00000Services(KaizenUser);
            return Json(service.Delete(CM00000), JsonRequestBehavior.AllowGet);
        }
        public ActionResult GetSingle(string KaizenPublicKey, string CompanyID)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            if (KaizenUser == null) return RedirectToAction("LoginToCompany", "SysUser");
            CM00000Services service = new CM00000Services(KaizenUser);
            CM00000 L = service.GetSingle(CompanyID);
            return Json(L, JsonRequestBehavior.AllowGet);
        }
        public ActionResult FillDropDownList(string KaizenPublicKey)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            CM00000Services service = new CM00000Services(KaizenUser);
            IList<CM00000> L = service.GetAll();
            return Json(L, JsonRequestBehavior.AllowGet);
        }
        public ActionResult GetStandardFields(string KaizenPublicKey)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            CM00029Services service = new CM00029Services(KaizenUser);
            IList<CM00038> o = service.GetStandardFields();
            return Json(o, JsonRequestBehavior.AllowGet);
        }
        public ActionResult UpdateStandardFields1(IList<CM00038> CM00038List, string KaizenPublicKey)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            if (KaizenUser == null) return RedirectToAction("LoginToCompany", "SysUser");
            CM00029Services service = new CM00029Services(KaizenUser);
            IList<CM00038> standardFields = service.GetStandardFields();

            foreach (CM00038 field in standardFields)
            {
                foreach (CM00038 updateField in CM00038List)
                {
                    if (field.FieldName.Equals(updateField.FieldName, StringComparison.OrdinalIgnoreCase))
                    {
                        updateField.IsEmailable = field.IsEmailable;
                        updateField.IsFilterable = field.IsFilterable;
                        updateField.Islockable = field.Islockable;
                        updateField.Islocked = field.Islocked;
                        updateField.IsPrintable = field.IsPrintable;
                        updateField.IsRequired = field.IsRequired;
                        updateField.IsSortable = field.IsSortable;
                        updateField.IsActive = field.IsActive;
                    }
                }
            }

            return Json(service.UpdateCM00038(CM00038List), JsonRequestBehavior.AllowGet);
        }

        public ActionResult UpdateStandardFields2(IList<CM00038> CM00038List, string KaizenPublicKey)
        {
            Kaizen.Configuration.KaizenSession KaizenUser = UserServices.GetKaizenSession(KaizenPublicKey);
            if (KaizenUser == null) return RedirectToAction("LoginToCompany", "SysUser");
            CM00029Services service = new CM00029Services(KaizenUser);
            IList<CM00038> standardFields = service.GetStandardFields();

            foreach (CM00038 field in standardFields)
            {
                foreach (CM00038 updateField in CM00038List)
                {
                    if (field.FieldName.Equals(updateField.FieldName, StringComparison.OrdinalIgnoreCase))
                    {
                        updateField.FieldDisplay = field.FieldDisplay;
                        updateField.FieldName = field.FieldName;
                        updateField.FieldOrder = field.FieldOrder;
                        updateField.FieldWidth = field.FieldWidth;
                        updateField.FieldTypeID = field.FieldTypeID;
                    }
                }
            }
            
            return Json(service.UpdateCM00038(CM00038List), JsonRequestBehavior.AllowGet);
        }
    }
}

File: UIServer/ScriptUI/CMS/Set_SetUpController.js
app.controller('Set_SetUpController', function ($scope, $http) {
    $scope.CM00000 = { Status: 1 };
    $scope.PagesCM00000 = [];
    $scope.LoadLookUp = function () {
        $http.get('/Set_SetUp/GetSingle?KaizenPublicKey=' + sessionStorage.PublicKey + "&SetupID=" + 1).success(function (data) {
            if (data.SetupID == 1) {
                $scope.CM00000 = data;
                $scope.CM00000.Status = 2;
            }
            else {
                $scope.CM00000 = { Status: 1 };
            }
        }).finally(function () { });
    };
    $scope.LoadLookUp();

    $scope.SaveData = function () {
        $http.post('/CMS_Set_CM00000/SaveData', { 'CM00000': $scope.CM00000, 'KaizenPublicKey': sessionStorage.PublicKey }).success(function (data) {
            if (data.Status == true) {
                $.bigBox({
                    title: data.Massage,
                    content: data.Description,
                    color: "#739E73",
                    timeout: 4000,
                    icon: "fa fa-check shake animated"
                });
                $scope.LoadLookUp();
            }
            else {
                $.bigBox({
                    title: data.Massage,
                    content: data.Description,
                    color: "#C46A69",
                    timeout: 4000,
                    icon: "fa fa-bolt swing animated"
                });
            }

        }).error(function (data, status, headers, config) {
            $.bigBox({
                title: data.Massage,
                content: data.Description,
                color: "#C46A69",
                timeout: 4000,
                icon: "fa fa-bolt swing animated"
            });
        });
    };
    $scope.UpdateData = function () {
        $http.post('/CMS_Set_CM00000/UpdateData', { 'CM00000': $scope.CM00000, 'KaizenPublicKey': sessionStorage.PublicKey }).success(function (data) {
            if (data.Status == true) {
                $.bigBox({
                    title: data.Massage,
                    content: data.Description,
                    color: "#739E73",
                    timeout: 4000,
                    icon: "fa fa-check shake animated"
                });
                $scope.LoadLookUp();
            }
            else {
                $.bigBox({
                    title: data.Massage,
                    content: data.Description,
                    color: "#C46A69",
                    timeout: 4000,
                    icon: "fa fa-bolt swing animated"
                });
            }

        }).error(function (data, status, headers, config) {
            $.bigBox({
                title: data.Massage,
                content: data.Description,
                color: "#C46A69",
                timeout: 4000,
                icon: "fa fa-bolt swing animated"
            });
        });
    };

    $scope.Cancel = function () {
        $scope.Clear();
        $scope.PagesCM00000 = [];
    };

    $scope.Clear = function () {
        $scope.StandardFieldsList = [];
    };


    $scope.StandardFields = function() {
        $scope.LoadCM00000Page('StandardFields');
        //$scope.Clear();
        $scope.LoadStandardFields();
    };
    $scope.LoadCM00000Page = function (ActionName) {
        removeEntityPage($scope.PagesCM00000);
        var URIPath = "/CMS/" + $scope.ToolBar.ActiveScreen.ID + "/" + ActionName + "?KaizenPublicKey=" + sessionStorage.PublicKey;
        var Page = { url: URIPath, ActionName: ActionName };
        $scope.PagesCM00000.push(Page);
    }
    $scope.StandardFieldsList = [];
    $scope.LoadStandardFields = function () {
        $http.get('/Set_SetUp/GetStandardFields?KaizenPublicKey=' + sessionStorage.PublicKey).success(function (data) {
                if (data.length >= 0) {
                    if (data.length > 0)
                        $scope.StandardFieldsList = data;
                    else
                        $scope.StandardFieldsList = [];
                }
            });
    };
    $scope.UpdateStandardFields = function () {
        //var selView = $scope.CM00071.SelectedView;
        $scope.fieldList1 = [];
        $scope.StandardFieldsList.forEach(function (ele, index) {
            var temp = {
                FieldDisplay: ele.FieldDisplay,
                FieldName: ele.FieldName,
                FieldOrder: ele.FieldOrder,
                FieldWidth: ele.FieldWidth,
                //IsActive: ele.IsActive,
                //IsEmailable: ele.IsEmailable,
                //IsFilterable: ele.IsFilterable,
                //Islockable: ele.Islockable,
                //Islocked: ele.Islocked,
                //IsPrintable: ele.IsPrintable,
                //IsRequired: ele.IsRequired,
                //IsSortable: ele.IsSortable
                FieldTypeID: ele.FieldTypeID
            }
            $scope.fieldList1.push(temp);
        });
        $http.post('/Set_SetUp/UpdateStandardFields1', {
            'CM00038List': $scope.fieldList1,'callNumber':1,
            'KaizenPublicKey': sessionStorage.PublicKey
        }).success(function (data) {
            if (data.Status == true) {
                $.bigBox({
                    title: data.Massage,
                    content: data.Description,
                    color: "#739E73",
                    timeout: 8000,
                    icon: "fa fa-check shake animated"
                });

                $scope.fieldList2 = [];
                $scope.StandardFieldsList.forEach(function (ele, index) {
                    var temp = {
                        //FieldDisplay: ele.FieldDisplay,
                        FieldName: ele.FieldName,
                        //FieldOrder: ele.FieldOrder,
                        //FieldWidth: ele.FieldWidth,
                        IsActive: ele.IsActive,
                        IsEmailable: ele.IsEmailable,
                        IsFilterable: ele.IsFilterable,
                        Islockable: ele.Islockable,
                        Islocked: ele.Islocked,
                        IsPrintable: ele.IsPrintable,
                        IsRequired: ele.IsRequired,
                        IsSortable: ele.IsSortable,
                        //FieldTypeID: ele.FieldTypeID
                    }
                    $scope.fieldList2.push(temp);
                });
                $http.post('/Set_SetUp/UpdateStandardFields2', {
                    'CM00038List': $scope.fieldList2, 'callNumber': 2,
                    'KaizenPublicKey': sessionStorage.PublicKey
                }).success(function (data) {
                    if (data.Status == true) {
                        $.bigBox({
                            title: data.Massage,
                            content: data.Description,
                            color: "#739E73",
                            timeout: 8000,
                            icon: "fa fa-check shake animated"
                        });
                    }
                }).error(function (data, status, headers, config) {
                    alert(data);
                });
            }
            else {
                //alert('ssssss')
            }
        }).error(function (data, status, headers, config) {
            alert(data);
        });
    };
});